src/client-side-encryption/state_machine.ts

```diff
@@ -524,9 +525,7 @@ export class StateMachine {
       .listCollections(filter, {
         promoteLongs: false,
         promoteValues: false,
-        ...(timeoutContext?.csotEnabled()
-          ? { timeoutMS: timeoutContext?.remainingTimeMS, timeoutMode: 'cursorLifetime' }
-          : {})
+        timeoutContext: timeoutContext && new CursorTimeoutContext(timeoutContext, Symbol())
```

Comment:
Neat! I didn't know you could `&&` objects together.

Reply:
Yup - `&&` and `||` return the first falsy/truthy value they encounter: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND

`timeoutContext` is either a timeout context or undefined, so if it is undefined, `timeoutContext && new CursorTimeoutContext(timeoutContext, Symbol())` returns `undefined`.  Otherwise, we proceed to `new CursorTimeoutContext(timeoutContext, Symbol())`, which is truthy, so it is returned.